crawl-ref/source/monster.cc

```diff
@@ -6347,6 +6347,14 @@ bool monster::is_web_immune() const
             || mons_genus(type) == MONS_MOTH
             || mons_genus(type) == MONS_DEMONIC_CRAWLER
             || is_insubstantial()
+            || type == MONS_EMPEROR_SCORPION
+            || type == MONS_SPARK_WASP
+            || type == MONS_STEELBARB_WORM
+            || type == MONS_TORPOR_SNAIL
+            || type == MONS_RADROACH
+            || type == MONS_BOULDER_BEETLE
+            || type == MONS_JOROGUMO
+            || mons_genus(type) == MONS_KILLER_BEE
             || mons_genus(type) == MONS_JELLY;
```

Comment:
This is awful. Can we add a new flag for this?

I recommend wandering into `mon-flags.h` and adding `M_WEB_IMMUNE = BIT(15)` on line 59, adding it to *at least* the monsters explicitly listed by type here, and then checking it here with `mons_class_flag(type, M_WEB_IMMUNE)`. As a more advanced move, also add it to the non-dummy genus monsters listed here (`MONS_JELLY` etc) and also check `|| mons_class_flag(mons_genus(type), M_WEB_IMMUNE)`.

If you'd rather not do this, please remind me when you merge this branch and I'll make the change myself.